Linking fails when building with CMAKE_USE_LIBBPF_PACKAGE=yes and ENABLE_USDT=OFF

Building latest bcc with `-DCMAKE_USE_LIBBPF_PACKAGE=yes` and `DENABLE_USDT=OFF` fails during linking of `test_static`.

I was initially trying to get it to build for i568 (which don't have USDT supporte, AFAIK), but it turns out that linking failure occurs of `x86_64` as well.

There's no error if I use `CMAKE_CMAKE_USE_LIBBPF_PACKAGE=no` instead.

```
$ cmake -DCMAKE_USE_LIBBPF_PACKAGE=yes -DPYTHON_CMD=python3 -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_LLVM_SHARED=1 -DENABLE_USDT=OFF .. && make -j8 VERBOSE=1
...
make  -f tests/cc/CMakeFiles/test_static.dir/build.make tests/cc/CMakeFiles/test_static.dir/build
make[2]: Entering directory '/home/somebody/git/bcc/build'
[ 99%] Building C object tests/cc/CMakeFiles/test_static.dir/test_static.c.o
cd /home/somebody/git/bcc/build/tests/cc && /usr/bin/cc  -I/usr/include/../tools/clang/include -I/home/somebody/git/bcc/src/cc -I/home/somebody/git/bcc/src/cc/api -I/home/somebody/git/bcc/src/cc/libbpf/include/uapi -I/home/somebody/git/bcc/tests/python/include -Wall -Wno-unused-result -fPIC -O3 -DNDEBUG -MD -MT tests/cc/CMakeFiles/test_static.dir/test_static.c.o -MF CMakeFiles/test_static.dir/test_static.c.o.d -o CMakeFiles/test_static.dir/test_static.c.o -c /home/somebody/git/bcc/tests/cc/test_static.c
[100%] Linking CXX executable test_static
cd /home/somebody/git/bcc/build/tests/cc && /usr/bin/cmake -E cmake_link_script CMakeFiles/test_static.dir/link.txt --verbose=1
/usr/bin/c++  -Wall  -Wno-unused-result -fPIC -O3 -DNDEBUG -rdynamic CMakeFiles/test_static.dir/test_static.c.o -o test_static  -Wl,-rpath,/home/somebody/git/bcc/build/src/cc ../../src/cc/libbcc.so.0.22.0 ../../src/cc/frontends/b/libb_frontend.a ../../src/cc/frontends/clang/libclang_frontend.a -Wl,--whole-archive -lclang-cpp /usr/lib64/libLLVM.so.12 -Wl,--no-whole-archive -lelf -lbpf 
/usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: ../../src/cc/libbcc.so.0.22.0: undefined reference to `bcc_usdt_new_frompath'
/usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: ../../src/cc/libbcc.so.0.22.0: undefined reference to `bcc_usdt_close'
/usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: ../../src/cc/libbcc.so.0.22.0: undefined reference to `bcc_usdt_new_frompid'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/cc/CMakeFiles/test_static.dir/build.make:104: tests/cc/test_static] Error 1
make[2]: Leaving directory '/home/somebody/git/bcc/build'
make[1]: *** [CMakeFiles/Makefile2:898: tests/cc/CMakeFiles/test_static.dir/all] Error 2
make[1]: Leaving directory '/home/somebody/git/bcc/build'
make: *** [Makefile:146: all] Error 2

```

The undefined references mentions belongs to functions in `src/cc/usdt/usdt.cc`, which are not built because `-DENABLE_USDT=OFF`.

My guess is that the linking error has something to do with `src/cc/link_all.cc`.

```
$ cd build && grep -e 'bcc_usdt_new_frompid' -R .
grep: ./src/cc/CMakeFiles/bcc-shared.dir/link_all.cc.o: binary file matches
grep: ./src/cc/libbcc.so.0.22.0: binary file matches
grep: ./src/cc/libbcc.so.0: binary file matches
grep: ./src/cc/libbcc.so: binary file matches
...
```

--- a/src/cc/CMakeLists.txt
+++ b/src/cc/CMakeLists.txt
@@ -89,6 +89,7 @@ set_target_properties(bcc-shared PROPERTIES VERSION ${REVISION_LAST} SOVERSION 0
 set_target_properties(bcc-shared PROPERTIES OUTPUT_NAME bcc)
 
 if(ENABLE_USDT)
+  add_definitions(-DEXPORT_USDT)
   set(bcc_usdt_sources usdt/usdt.cc usdt/usdt_args.cc)
   # else undefined
 endif()

--- a/src/cc/link_all.cc
+++ b/src/cc/link_all.cc
@@ -13,9 +13,11 @@ namespace {
       if (::getenv("bar") != (char *)-1)
         return;
 
+#ifdef EXPORT_USDT
       (void)bcc_usdt_new_frompid(-1, nullptr);
       (void)bcc_usdt_new_frompath(nullptr);
       (void)bcc_usdt_close(nullptr);
+#endif
     }
   } LinkAll;  // declare one instance to invoke the constructor
 }

